cmd/internal/cli/push.go

```diff
@@ -34,6 +45,33 @@ var PushCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		// Push to library requires a valid authToken
 		if authToken != "" {
+			if !unauthenticatedPush {
+				// check if the container is signed
+				imageSigned, err := signing.IsSigned(args[0], "", 0, false, authToken, true)
+				if err != nil {
+					// warning message will be: ("Unable to verify the container (test.sif): %v", err)
+					sylog.Warningf("%v", err)
+				}
+				// if its not signed, print a warning
+				if !imageSigned {
+					sylog.Infof("TIP: Learn how to sign your own containers here : https://www.sylabs.io/guides/3.0/user-guide/signNverify.html")
+					fmt.Fprintf(os.Stderr, "\n")
+					sylog.Warningf("Your container is **NOT** signed! You REALLY should sign your container before pushing!")
+					fmt.Fprintf(os.Stderr, "Do you really want to continue? [N/y] ")
```

Comment:
can you please move lines 60-66 to a separate function? It should be the same as the other case, so you can reuse that code.